parsers/FR.py

```diff
@@ -1,92 +1,85 @@
-                data = datas[datetime]
-
-                if key == 'Hydraulique':
-                    # Hydro is a special case!
-                    if granularite == 'Global':
-                        continue
-                    elif granularite in ['FEE', 'LAC']:
-                        if not MAP_GENERATION[key] in data['production']:
-                            data['production'][MAP_GENERATION[key]] = 0
-                        # Run of the river or conventional
-                        data['production'][MAP_GENERATION[key]] += float(
-                            value.text)
-                    elif granularite == 'STT':
-                        if not MAP_STORAGE[key] in data['storage']:
-                            data['storage'][MAP_STORAGE[key]] = 0
-                        # Pumped storage generation
-                        data['storage'][MAP_STORAGE[key]] += -1 * float(value.text)
-                elif granularite == 'Global':
-                    if key in MAP_GENERATION:
-                        data['production'][MAP_GENERATION[key]] = float(value.text)
-                    elif key in MAP_STORAGE:
-                        if not MAP_STORAGE[key] in data['storage']:
-                            data['storage'][MAP_STORAGE[key]] = 0
-                        data['storage'][MAP_STORAGE[key]] += -1 * float(value.text)
-
-    return list(datas.values())
+    formatted_from = to.shift(days=-1).format('YYYY-MM-DDTHH:mm')
+    formatted_to = to.format('YYYY-MM-DDTHH:mm')
+
+    params = {
+        'apikey': '38c5e47f70c12c9f2e2e68937d8d2065e5c4c2f4f846e127dca0f06f',
+        'dataset': 'eco2mix-national-tr',
+        'q': 'date_heure >= {} AND date_heure <= {}'.format(
+            formatted_from, formatted_to),
+        'timezone': 'Europe/Paris',
+        'rows': 100
+    }
+
+    # make request and create dataframe with response
+    response = r.get(API_ENDPOINT, params=params)
+    data = json.loads(response.content)
+    data = [d['fields'] for d in data['records']]
+    df = pd.DataFrame(data)
+
+    # filter out desired columns and convert values to float
+    value_columns = MAP_GENERATION.keys() + MAP_HYDRO
+    df = df[['date_heure'] + value_columns]
+    df[value_columns] = df[value_columns].astype(float)
+
+    datapoints = list()
+    for row in df.iterrows():
+        production = dict()
+        for key, value in MAP_GENERATION.items():
+            production[value] = row[1][key]
+
+        # Hydro is a special case!
+        production['hydro'] = row[1]['hydraulique_lacs'] + row[1]['hydraulique_fil_eau_eclusee']
+        storage = {
+            'hydro': row[1]['pompage'] * -1 + row[1]['hydraulique_step_turbinage'] * -1
+        }
+
+        datapoints.append({
+            'zoneKey': zone_key,
+            'datetime': arrow.get(row[1]['date_heure']).datetime,
+            'production': production,
+            'storage': storage,
+            'source': 'https://opendata.reseaux-energies.fr'
```

Comment:
we normally only put the domain here. Can you remove "https://" ?

Reply:
Sure, will I also update the README?

Reply:
Ah yes that's probably a good idea.